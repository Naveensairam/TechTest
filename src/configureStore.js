import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';  

let composeEnhancers = compose;

if (__DEV__) {
  composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION__ || compose;
}

const configureStore = () => {
  return createStore(
    rootReducer,
    applyMiddleware(thunk)
    // composeEnhancers(applyMiddleware(thunk))
  );
};

export default configureStore;