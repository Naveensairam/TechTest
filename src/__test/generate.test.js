import React from 'react';
import { shallow } from 'enzyme';
import Generate from '../generate';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const store = mockStore();

const props = {};

describe('My Generate Component', () => {
    it('should render correctly with default props and redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Generate {...props} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
it('should dispatch both', () => {
    const dispatch = jest.fn();
    // store.dispatch = dispatch;
    dispatch();
    const wrapper = shallow(
        <Provider store={store}>
            <Generate {...props} error={true} />
        </Provider>);
        
   
    expect(dispatch).toHaveBeenCalledTimes(1);
    });