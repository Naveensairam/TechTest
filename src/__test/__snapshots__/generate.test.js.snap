// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Generate Component should render correctly with default props and redux store 1`] = `ShallowWrapper {}`;

exports[`My Tabs Component should render correctly with default props and redux store 1`] = `ShallowWrapper {}`;
